packages/perf/lib/index.js

```diff
@@ -80,6 +81,21 @@ class FirebasePerfModule extends FirebaseModule {
     return trace.start().then(() => trace);
   }
 
+  newScreenTrace(identifier) {
+    if (!isString(identifier) || identifier.length > 100) {
+      throw new Error(
+        "firebase.perf().newScreenTrace(*) 'identifier' must be a string with a maximum length of 100 characters.",
+      );
+    }
+
+    return new ScreenTrace(this.native, identifier);
+  }
+
+  startScreenTrace(identifier) {
```

Comment:
Note for @russellwheatley - this PR is from before the modular PR work and was in flight while you have been working. It adds two native-only APIs that need to go in the #6771 API after this is merged (should merge as soon as CI goes green) so this PR is no longer orphaned and modular/future API doesn't drop it in the shuffle